delivery_gls_asm/models/gls_asm_request.py

```diff
@@ -294,12 +294,12 @@ def _send_shipping(self, vals):
         _logger.debug(res)
         res["_return"] = int(res["Resultado"]["_return"])
         if res["_return"] < 0:
-            raise UserError(_(
-                "GLS returned an error trying to record the shipping for {}.\n"
-                "Error:\n{}").format(
-                    vals.get("referencia_c", ""),
-                    res.get("Errores", {}).get(
-                        "Error", "code {}".format(res["_return"]))))
+            raise UserError(
+                _(
+                    "GLS returned an error trying to record the shipping for {}.\n"
+                    "Error:\n{}"
+                ).format(vals.get("referencia_c", ""), res["_return"])
```

Comment:
Check upper versions for a more proper fix:

https://github.com/OCA/l10n-spain/blob/ade52b7325e15a46908734e313204a7eaff34c5e/delivery_gls_asm/models/gls_asm_request.py#L259-L270